### PR TITLE
Phosphor upgrade

### PR DESCRIPTION
**Pull request by blink1073**
Upgrade notes:

- Most of the path changeovers were made using a variant of:
```
find . -name "*.ts" -exec sed -i '' -e 's/phosphor\/lib\/core\/token/@phosphor\/application/g' {} +
```
- The bulk of the manual work was in conforming to the new `Signal` and `ReadonlyArray` APIs.  Both resulted in cleaner code and APIs in JupyterLab.
- Added `common/vector.ts` for now to preserve `ObservableVector`.
- No changes to Phosphor were required for this PR.

### CODE REVIEW

File: src/completer/widget.ts [hunk 1]
```diff
@@ -406,7 +406,7 @@ class CompleterWidget extends Widget {
     let items = this.node.querySelectorAll(`.${ITEM_CLASS}`);
     let subset = Private.commonSubset(Private.itemValues(items));
     let query = this.model.query;
-    if (subset && subset !== query && subset.indexOf(query) === 0) {
+    if (subset && subset !== query && subset.ArrayExt.firstIndexOf(query) === 0) {
```
Comment (jasongrout) [inline src/completer/widget.ts]: `ArrayExt.firstIndexOf(subset, query)`?

Comment (jasongrout) [inline src/completer/widget.ts]: or just revert the change

File: src/filebrowser/crumbs.ts [hunk 2]
```diff
@@ -271,7 +263,7 @@ class BreadCrumbs extends Widget {
         if (error.xhr) {
           error.message = `${error.xhr.status}: error.statusText`;
         }
-        if (error.message.indexOf('409') !== -1) {
+        if (error.message.ArrayExt.firstIndexOf('409') !== -1) {
```
Comment (jasongrout) [inline src/filebrowser/crumbs.ts]: Another `.indexOf` that should probably have been left alone.

**Comment by jasongrout** [comment]
Awesome, thanks! Searching the Changed Files tab for `.indexOf` shows a few places where there probably should not have been a substitution.

File: src/filebrowser/crumbs.ts (hunk 2 above)
Comment (jasongrout) [inline src/filebrowser/crumbs.ts]: (There are a few more here, if you search for `.indexOf`, or in the new files, search for `.ArrayExt.firstIndexOf`)

Comment (blink1073) [inline src/filebrowser/crumbs.ts]: Yeah, the compiler is catching these for us.  Whoops.

**Comment by jasongrout** [comment]
nice, that was fast! I was dreading this huge (line number) change :).
